src/core/network-openssl.c

```diff
@@ -656,13 +661,20 @@ static void set_pubkey_info(TLS_REC *tls, X509 *cert, unsigned char *cert_finger
 	bio = BIO_new(BIO_s_mem());
 	ASN1_TIME_print(bio, X509_get_notAfter(cert));
 	length = BIO_read(bio, buffer, sizeof(buffer));
+	if (length < 0) {
+		ret = 0;
+		goto done;
+	}
 	buffer[length] = '\0';
 	BIO_free(bio);
 	tls_rec_set_not_after(tls, buffer);
 
+done:
 	g_free(cert_fingerprint_hex);
 	g_free(public_key_fingerprint_hex);
 	EVP_PKEY_free(pubkey);
+
+	return ret ? TRUE : FALSE;
```

Comment:
...and just "return ret" here

(the ternary makes sense in the other function that may return -1 or 0, but this one is straight gboolean)